client/src/controllers/DrilldownController.test.js

```diff
@@ -0,0 +1,211 @@
+import { Application } from '@hotwired/stimulus';
+import { DrilldownController } from './DrilldownController';
+import { DropdownController } from './DropdownController';
+import { ActionController } from './ActionController';
+
+jest.useFakeTimers();
+
+describe('DrilldownController', () => {
+  const eventNames = [
+    'w-drilldown:opened',
+    'w-drilldown:closed',
+    'w-drilldown:ready',
+    'w-drilldown:toggled',
+  ];
+
+  const events = {};
+  let application;
+  let errors = [];
+
+  const setup = async (
```

Comment:
OK, I think the HTML still needs some work, it's super hard to parse/read in it's current format & nesting, here's a refined version that will be used for the other fixes below.

Observe we are now making some (not all) of the attributes drop to new lines, we have added some `id` attributes that make it easier to grab elements and have used a bit of nested JS to make the `data-action` items easier to read.

```js
  const setup = async (
    html = `
    <section>
      <div
        id="filters-drilldown"
        class="w-drilldown"
        data-controller="w-drilldown"
        data-action='${[
          'w-swap:success@document->w-drilldown#updateCount',
          'w-dropdown:hide->w-drilldown#delayedClose',
          'w-dropdown:clickaway->w-drilldown#preventOutletClickaway',
        ].join(' ')}'
        data-w-drilldown-count-attr-value="data-w-active-filter-id"
      >
        <span class="w-drilldown__count" data-w-drilldown-target="count"></span>
        <button type="button">Show Filters</button>
        <div class="w-drilldown__menu" data-w-drilldown-target="menu" hidden>
          <h2>Filter by</h2>
          <button
            id="drilldown-toggle"
            type="button"
            aria-expanded="false"
            aria-controls="drilldown-field-0"
            data-action="click->w-drilldown#open"
            data-w-drilldown-target="toggle"
          >
            Field 1
          </button>
        </div>
        <div class="w-drilldown__submenu" id="drilldown-field-0" hidden tabIndex="-1">
          <button class="w-drilldown__back" type="button" data-action="click->w-drilldown#close">Back</button>
        </div>
      </div>
      <button type="button" id="action-element" data-controller="w-action">Action</button>
    </section>`,
    identifier = 'w-drilldown',
  ) => {
    document.body.innerHTML = `<main>${html}</main>`;
```